Sources/VLCEditToolbar.swift

```diff
@@ -13,30 +13,89 @@ protocol VLCEditToolbarDelegate: class {
     func delete()
     func createPlaylist()
     func rename()
+    func share()
 }
 
-// Decided to use a UIView instead of UIToolbar because we have more freedom
-// FIXME: Basic structure without UI
 class VLCEditToolbar: UIView {
     weak var delegate: VLCEditToolbarDelegate?
+    private var category: MediaLibraryBaseModel
```

Comment:
I might be mistaken here, it seems that the category is only used for file type detection.
What would you think about using only the file type to avoid sending unused information?
Something like:

```Swift

private var fileType: VLCMLObject.Type

...

if !(fileType is VLCMLAlbum.Type) && !(file is VLCMLArtist.Type) && !(file is VLCMLGenre.Type) { ... }

```

Reply:
actually that filetype detection currently doesn't even work :/ I need to think of something better

Reply:
I looked at the code and it actually it works. I will probably have to adapt it for the allFiles tab at some point. I believe there we want to disable the button as soon as the selected cells contain one of the above mentioned file types. Which can change with every selection. So I believe I need to keep the category here and iterate over all selected files.

Reply:
I'll make a ticket for it